### PR TITLE
Translate adventures to German

### PR DESCRIPTION
**Pull request by DerSibi**
Start translating adventures to German

**Description**

I create the language file for the German translation of adventures and translate the first two level of the "story" adventure.
I specifically open the PR early because I realize that @krulls is working on German translations as well and want to avoid double work. I hope it's fine that way - in other cases I'm open for suggestions how to proceed.

**Fix for**

Describe what issue or discussion your pr concerns

**How to test**

Unit tests pass, YAML schema is configured to be checked against as requested in contribution guidelines.

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] Links to an existing issue or discussion (if not, create an issue first)
- [ ] Describes changes clear in the format above (present tense, no subject)
- [ ] Has a "how to test" section
- [ ] Only one thing is done in this pull request (specifically please try to refrain from mixing textual changes to the yamls from code changes)

### CODE REVIEW

**Comment by Felienne** [comment]
Thanks @DerSibi!! 

All tests pass so the code is correct in all snippets. Glancing at it, the translations look good to me, but beware I don't speak german well. Maybe @dpprdan, @krulls or @nichtich want to take a peek?

**Comment by DerSibi** [comment]
Hey @Felienne,

thanks! Indeed, feedback would be appreciated :) I'm not so super familiar with how things work here on github. Should I invite reviewers (and if so: how?). Also, should I mark this PR as draft until at least the first adventure is completely translated?

**Comment by Felienne** [comment]
> thanks! Indeed, feedback would be appreciated :) I'm not so super familiar with how things work here on github. 

You are doing quite well for someone who it not familiar so far! 

>Should I invite reviewers (and if so: how?). 

Great questions! At the top right you can invite reviewers:

image

Often GitHub suggests reviewers familiar with the files you have edited. But you can also tag people (as you already did with krulls in your first message) depending on their settings people will get an email notification from that.

> Also, should I mark this PR as draft until at least the first adventure is completely translated?

We typically use draft from things that are not finished yet (meaning: where the author still expects to do work) So your is not a draft since you think it is good to go! Let's see if the Germans have feedback, if not, I feel confident merging in a few days, since we test all code snippets so we are at least sure that it all will work, and it seems that you did a good job judging with my high-school level German

File: coursedata/adventures/de.yaml [hunk 1]
```diff
@@ -86,95 +86,102 @@ adventures:
...
-                            print 'Robin leaves the shop and goes home'
+                            print 'Robin verlässt das Geschäft und geht nach Hause.'
                     else
-                        print 'Robin goes home'
+                        print 'Robin geht nach Hause.'
                     ```
 
-                    ## More elaborate
-                    The example above is pretty straightforward, but with nesting you can really elaborate your code and turn your story into a real game. Check out this example!
+                    ## Etwas komplizierter
```
Comment (krulls) [inline coursedata/adventures/de.yaml:109]:
```suggestion
                    ## Etwas ausführlicher
```

**Review by krulls — APPROVED** [review]
looks good from my point of view. not sure if you want to continue in this branch, anyway I put an approval already.

**Comment by DerSibi** [comment]
> You are doing quite well for someone who it not familiar so far!

Thanks :)

> Great questions! At the top right you can invite reviewers:

I don't have that gear icon there - I suppose that means I'm not allowed to add reviewers on my own?

Screenshot_20220128_185548

> We typically use draft from things that are not finished yet (meaning: where the author still expects to do work) So your is not a draft since you think it is good to go! Let's see if the Germans have feedback, if not, I feel confident merging in a few days, since we test all code snippets so we are at least sure that it all will work, and it seems that you did a good job judging with my high-school level German

Alright. Then I'll leave it like that. Last question: Looks like I'll finish the story adventure soon. Should I keep working on this branch or open a new branch (and an independent PR) e.g. with each adventure?

File: coursedata/adventures/de.yaml [hunk 2]
```diff
@@ -0,0 +1,2478 @@
...
+
+                    ## Hedy Beispiel-Code
+
+                    ```
+                    name is ask Wie lautet der Name der Hauptfigur?
+                    print name läuft nun durch einen dunklen Wald.
+                    print name hat etwas Angst.
+                    print Plötzlich ertönen seltsame Geräusche...
+                    sleep
+                    print name hat Angst. Schließlich spukt es in diesem Wald.
```
[the remark this answers is not in the document]
Comment (DerSibi) [reply coursedata/adventures/de.yaml:54]: Ahhh... Yes. I mistook it for a missing full stop. I'll change it.

File: coursedata/adventures/de.yaml [hunk 3]
```diff
@@ -0,0 +1,2478 @@
...
+                    ```
+                start_code: "print Deine Geschichte beginnt genau hier."
+            12:
```
[the remark this answers is not in the document]
Comment (DerSibi) [reply coursedata/adventures/de.yaml:57]: Frankly, I don't know either. I just copied the EN language file and started translating it pretty much line by line. You think that this is a problem to have the rest (still) in English?

File: coursedata/adventures/de.yaml (hunk 1 above)
Comment (DerSibi) [reply coursedata/adventures/de.yaml:109]: Agreed.

**Comment by Felienne** [comment]
> I don't have that gear icon there - I suppose that means I'm not allowed to add reviewers on my own?
> 
> Screenshot_20220128_185548

Ah maybe that exists for contributors only, let me invite you to the repo so you can do it do (you should get an email soon) That will also allow you to create branches in this repo which is a bit easier for us because we could then push small fixes to it.

> Alright. Then I'll leave it like that. Last question: Looks like I'll finish the story adventure soon. Should I keep working on this branch or open a new branch (and an independent PR) e.g. with each adventure?

When possible, smaller PR's are preferred because the make checks and merging easier. You might see that in the future other people also change files (I have made some changes today to German yamls) and then those changes will need to be reconciled (=merging).

**Review by Felienne — APPROVED** [review]

File: coursedata/adventures/de.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Felienne) [reply coursedata/adventures/de.yaml:57]: No it is no problem! The implementation picks up the English version if you would remove it, so the view for the user would be the same.

The only reason to remove it it that the English version of the yaml can get outdated (I made changes today, so it is outdated!) so please remember to copy it again when you continue.

**Pull request merged**